### DISCUSSION TITLE
Geosearch naming: keep or not `_` in suffix?

### DISCUSSION THREAD

**Discussion by curquiza** [meilisearch/product#441] [Feedback & Feature Proposal]
Following this discussion: https://github.com/meilisearch/product/discussions/439

For v0.28.0, we could consider this breaking naming change.

Is it relevant to keep or remove the `_` in front of different geosearch fields?
- `_geo`: in the documents themself
- `_geoRadius`/`_geoPoint`: used in the search parameter `filter`/`sort`

@bidoubiwa opinion (in the previous linked discussion)

> I don't think this usage is easy to understand or to use. I also understand why we added the _ in front of the function name, but it adds a layer of possible mistakes and confusion. With the function (or what appears to be one) and the _ it feels a bit like calling a private method in a string. I'm not sure we want to give the sensation to the user that he is coding in a string.

@gmourier opinion (in the previous linked discussion)
> Nevertheless, I'm not directly attached to the fact of including _ in the name of the built-in functions or _geo, I don't know if it's better or worse, it deserves a dedicated discussion.

@curquiza opinion: the breaking should be worth it! If it's breaking for "nothing" or almost, we don't break 😇 

What is your opinion on this?

**Comment by curquiza** [comment]
For additional context: no user complained/reported about the `_` suffix so far. However, this is still something we could consider :) 
As usual, the breaking should be worth it.

**Comment by irevoire** [comment]
In my opinion, since no one ever complained, there is no reason to change it.
But adding the option to configure where is your `_geo` object located would be a great addition to the feature because some people complained they had to remap their whole dataset to fit our format.

Also, I think almost everyone uses `lon` for the longitude, and we use `lng`. Thus having an option to configure the path to the longitude and the latitude separately might be even better than letting people change the path of the `_geo` object.
With the support of the nested fields that should cover most use cases.

**Reply by curquiza** [reply]
Thanks for the feedback @irevoire! Since this discussion is only focused on naming and your suggestion could be a nice improvement, can I let you create another discussion for what you mention? 😇

**Reply by irevoire** [reply]
Yeah, well for the naming, in a word, I just wanted to say that it's useless to change it since it'll never work for everyone, and we should focus on the configuration of the `lat` / `lng`.

**Reply by curquiza** [reply]
Yes I got your idea in your previous message 😁 Just saying the improvement you suggested is nice as well and would deserve a dedicated discussion, but I will open it, I can understand you don't have the time

**Reply by curquiza** [reply]
New open discussion here: https://github.com/meilisearch/product/discussions/444

**Comment by curquiza** [comment]
Regarding the feedback @gmourier and I decided to keep the `_` as it is for v0.28.0, so probably (99% sure) for v1.
It does not mean the discussion cannot continue and could not be considered in another versions
